pandas/core/generic.py

```diff
@@ -3630,6 +3630,93 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
             return res
 
     # ----------------------------------------------------------------------
+    # Timeseries methods Methods
+
+    def asof(self, where, subset=None):
+        """
+        Return last good (non-NaN) value in object if subset is NaN for
+        requested date if the input is a Series
```

Comment:
"if subset is NaN for requested date if the input is a Series" is a bit a difficult sentence. I also think it is maybe not needed. The 'non-NaN' is already mentioned, and using 'subset' is a bit confusing, as this is not used for Series.